io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/internal/CSVInstanceWriter.java

```diff
@@ -97,44 +97,50 @@ protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 
 		// get all instances of the selected Type
 		InstanceCollection instances = getInstanceCollection(selectedTypeName);
-		Iterator<Instance> instanceIterator = instances.iterator();
-		Instance instance = null;
-		try {
-			instance = instanceIterator.next();
-		} catch (NoSuchElementException e) {
-			reporter.error(new IOMessageImpl("There are no instances for the selected type.", e));
-			return reporter;
-		}
+		// use ResourceIterator<Instance> in a try block because is closable -
+		// avoid infinite
+		// cleaning project after exporting data
+		try (ResourceIterator<Instance> instanceIterator = instances.iterator();) {
+			Instance instance = null;
+			try {
+				instance = instanceIterator.next();
+			} catch (NoSuchElementException e) {
+				reporter.error(
+						new IOMessageImpl("There are no instances for the selected type.", e));
+				return reporter;
+			}
 
-		// get definition of current instance (only this properties with this
-		// definition type will be written to csv file)
-		TypeDefinition definition = instance.getDefinition();
+			// get definition of current instance (only this properties with
+			// this
+			// definition type will be written to csv file)
+			TypeDefinition definition = instance.getDefinition();
 
-		// first csv file doesn't have a header row, so it' necessary to write
-		// it to a temp directory
-		File tempDir = Files.createTempDir();
-		File tempFile = new File(tempDir, "tempInstances.csv");
+			// first csv file doesn't have a header row, so it' necessary to
+			// write
+			// it to a temp directory
+			File tempDir = Files.createTempDir();
+			File tempFile = new File(tempDir, "tempInstances.csv");
 
-		// write instances to csv file (without header)
```

Comment:
The comment `// write instances to csv file (without header)`  got deleted in the new code. I guess it would be useful if added back.